compiler/p/codegen/PPCSystemLinkage.cpp

```diff
@@ -1717,7 +1717,8 @@ void TR::PPCSystemLinkage::setParameterLinkageRegisterIndex(TR::ResolvedMethodSy
                }
             numFloatArgs++;
             numIntArgs++;
-            if (paramCursor->getDataType() == TR::Double) numIntArgs++;
+            if (!TR::Compiler->target.is64Bit() && paramCursor->getDataType() == TR::Double)
+               numIntArgs++;
```

Comment:
Naive question. Why are numIntArgs being incremented when this is a double (and numFloatArgs has already been incremented) ?

Reply:
The ABI wants what it wants. I found it odd, too.

But incrementing it twice for a double on a 64-bit platform was even more odd ;) .

Reply:
This is correct for ppc64le, floating point args passed in registers cause the corresponding integer registers to be 'burned'.

Having said that you don't actually have to do anything for 32-bit targets because LE is 64-bit only (I realize there's another `is32Bit()` check in this code, it too is unnecessary and was probably copied from some other Power linkage that has both 32- and 64-bit targets).

Reply:
Interesting; I hadn't realized this function was PPCLE specific (there are lots of references to AIX and 32-bit in other spots in this class). I suppose you're recommending that I clear out the 32-bit code from this path then, @ymanton :) ?

Reply:
Yeah, the

```
if (paramCursor->getDataType() == TR::Double) numIntArgs++;
```

line definitely needs to go to make it correct, but the two added lines aren't necessary. Otherwise, the changes in this PR look fine to me. This class can be refactored properly later if/when it needs to support other platforms.

Reply:
Ok, I updated the commit to simply remove the second numIntArgs bump.

Thanks @ymanton !